webdriver/tests/bidi/script/evaluate/evaluate.py

```diff
@@ -0,0 +1,160 @@
+        exception.value.result)
+
+
+@pytest.mark.asyncio
+async def test_interact_with_dom(bidi_session, top_context):
+    result = await bidi_session.script.evaluate(
+        expression="'window.location.href: ' + window.location.href",
+        target=ContextTarget(top_context["context"]))
+
+    assert result == {
+        "type": "string",
+        "value": "window.location.href: about:blank"}
+
+
+@pytest.mark.asyncio
+async def test_resolved_promise_with_await_promise_false(bidi_session,
+                                                         top_context):
+    result = await bidi_session.script.evaluate(
+        expression="Promise.resolve('SOME_RESOLVED_RESULT')",
+        target=ContextTarget(top_context["context"]),
+        await_promise=False)
+
+    recursive_compare({
+        "type": "promise",
+        "handle": any_string},
+        result)
+
+
+@pytest.mark.asyncio
+async def test_resolved_promise_with_await_promise_true(bidi_session,
+                                                        top_context):
+    result = await bidi_session.script.evaluate(
+        expression="Promise.resolve('SOME_RESOLVED_RESULT')",
+        target=ContextTarget(top_context["context"]),
+        await_promise=True)
+
+    assert result == {
+        "type": "string",
+        "value": "SOME_RESOLVED_RESULT"}
+
+
+@pytest.mark.asyncio
+async def test_resolved_promise_with_await_promise_omitted(bidi_session,
+                                                           top_context):
+    result = await bidi_session.script.evaluate(
+        expression="Promise.resolve('SOME_RESOLVED_RESULT')",
+        target=ContextTarget(top_context["context"]))
+
+    assert result == {
+        "type": "string",
+        "value": "SOME_RESOLVED_RESULT"}
+
+
+@pytest.mark.asyncio
+async def test_rejected_promise_with_await_promise_false(bidi_session,
+                                                         top_context):
+    result = await bidi_session.script.evaluate(
+        expression="Promise.reject('SOME_REJECTED_RESULT')",
+        target=ContextTarget(top_context["context"]),
+        await_promise=False)
+
+    recursive_compare({
+        "type": "promise",
+        "handle": any_string},
+        result)
+
+
+@pytest.mark.asyncio
+async def test_rejected_promise_with_await_promise_true(bidi_session,
+                                                        top_context):
+    with pytest.raises(ScriptEvaluateResultException) as exception:
+        await bidi_session.script.evaluate(
+            expression="Promise.reject('SOME_REJECTED_RESULT')",
+            target=ContextTarget(top_context["context"]),
+            await_promise=True)
+
+    recursive_compare({
+        'realm': any_string,
+        'exceptionDetails': {
+            'columnNumber': any_int,
+            'exception': {'type': 'string',
+                          'value': 'SOME_REJECTED_RESULT'},
+            'lineNumber': any_int,
+            'stackTrace': {'callFrames': []},
```

Comment:
This should be `any_stacktrace`? (at least I get a stacktrace on Firefox and fail this assert for now)

Reply:
done